WalletWasabi.Gui/Rpc/WasabiJsonRpcService.cs

```diff
@@ -121,18 +121,27 @@ public object GetStatus()
 		}
 
 		[JsonRpcMethod("build")]
-		public string BuildTransaction(PaymentInfo[] payments, OutPoint[] coins, int feeTarget, string password = null)
+		public string BuildTransaction(PaymentInfo[] payments, OutPoint[] coins, int feeTarget = 0, decimal feeRate = 0, string password = null)
 		{
 			Guard.NotNull(nameof(payments), payments);
 			Guard.NotNull(nameof(coins), coins);
-			Guard.InRangeAndNotNull(nameof(feeTarget), feeTarget, 2, Constants.SevenDaysConfirmationTarget);
 			password = Guard.Correct(password);
+			
+			var argFeeRate = new FeeRate(feeRate);
+
+			static bool InRange<T>(IComparable<T> val, T min, T max) =>
+				val.CompareTo(min) >= 0 && val.CompareTo(max) <= 0;
+
+			var feeStrategy = (feeRate, feeTarget) switch
+			{
+				(_, 0) when InRange(argFeeRate, Constants.MinRelayFeeRate, Constants.AbsurdlyHighFeeRate) => FeeStrategy.CreateFromFeeRate(argFeeRate),
+				(0, _) when InRange(feeTarget, 2, Constants.SevenDaysConfirmationTarget) => FeeStrategy.CreateFromConfirmationTarget(feeTarget),
```

Comment:
Why 2 is the minimum?

Reply:
FeeTarget minimum is 2 because Bitcoin Core/Knots do not provide estimation for confirming in next block, they provide a range [2..1008]. We are using this minimum also in other parts of the code.

Reply:
ACK

Reply:
Can you add this as a constant there and also comment with Lucas's reasoning.